tests/test_xmdf.cpp

```diff
@@ -299,6 +299,27 @@ TEST( MeshXmdfTest, withReferenceTime )
   MDAL_CloseMesh( m );
 }
 
+TEST( MeshXmdfTest, unlockWhenClose )
+{
+  std::string tmpXmdf = tmp_file( "temp.xmdf" );
+  copy( test_file( "/xmdf/withReferenceTime/PTM_005_QGIS_Axis.xmdf" ), tmpXmdf );
+
+  ASSERT_TRUE( fileExists( tmpXmdf ) );
+
+  std::string path = test_file( "/xmdf/withReferenceTime/hydraul_006.2dm" );
+  EXPECT_EQ( MDAL_MeshNames( path.c_str() ), "2DM:\"" + path + "\"" );
+  MDAL_MeshH m = MDAL_LoadMesh( path.c_str() );
+  ASSERT_NE( m, nullptr );
+  MDAL_M_LoadDatasets( m, tmpXmdf.c_str() );
+  ASSERT_EQ( 12, MDAL_M_datasetGroupCount( m ) );
+
+  MDAL_CloseMesh( m );
+
+  deleteFile( tmpXmdf );
```

Comment:
This test properly tests it only on Windows, right? Also on Windows I fear it may fail, because windows may have some delay for garbage collection and freeing handles. I am wondering if it is not better approach to check the same? Like check with HDF library that all the handles are closed?

Reply:
> This test properly tests it only on Windows, right?


yes


> windows may have some delay for garbage collection and freeing handles

Not sure to understand, do you mean that the HDF lib will have some delay to freeing the handle under Windows? Or windows will have so delay to really delete the file?


> Like check with HDF library that all the handles are closed?

Check from the test with HDF lib, is it possible without the file handle?